build-system/tasks/helpers.js

```diff
@@ -325,9 +325,30 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
   const babelCaller =
     options.babelCaller ?? (options.minify ? 'minified' : 'unminified');
 
+  const dependsOnRuntime = !!options.wrapper
+    ?.replace(/[\n\s]+/g, '')
+    .startsWith('(self.AMP=self.AMP||[]).push({');
```

Comment:
The idea here is that source code always imports the real URLs so that they always work.

When compiling, we know that the bundle depends on the runtime based on its wrapper, allowing us to do this optimization.

I'm not super happy with the way I'm accomplishing this, so I'm open to feedback if there's a cleaner way to do this.

Reply:
There's got to be a cleaner way to separate extensions from entry points.

Reply:
The wrapper is a function that returns a string. We could instead resolve it later (at this level), and compare the wrapper against the `wrappers.extension` function here. But I have no better idea than that.

Reply:
Is `options.wrapper` set to `'extension'`?

Reply:
At a higher level it is, but it's evaluated to the function result before it reaches this codepath. We could probably move the evaluation here.

Reply:
Maybe it's cleaner to pass a `babelPlugins` option from `buildExtensionJs`. PTAL.